src/core/core/rpc_engine.cpp

```diff
@@ -846,14 +850,28 @@ namespace dsn {
                             {
                                 resolver->on_access_failure(req2->header->gpid.u.partition_index, err);
 
+                                /*
+                                // TODO(qinzuoyan): because the user will fetch reply states from this response_task 'call',
```

Comment:
Your understanding is not right unfortunately. Here we are already running inside the given response_task, and user provided callback is replaced and given as ```callback``` here, which is why we can start a new rpc-response-task.